test/cypress/integration/settings/your-psp-stripe-tasklist.cy.test.js

```diff
@@ -175,4 +176,30 @@ describe('Your PSP Stripe page', () => {
       cy.get('strong[id="task-vatNumber-status"]').should('contain', 'complete')
     })
   })
+
+  describe('Company number task', () => {
+    it('should click company registration task and display page correctly', () => {
+      setupYourPspStubs()
+      cy.get('span').contains('Company registration number').click()
+      cy.get('h1').should('contain', 'Company registration number')
+    })
+
+    it('should redirect back to the task List when a valid company registration number is submitted', () => {
+      setupYourPspStubs()
+
+      cy.get('#company-number-declaration').click()
+      cy.get('#company-number').type(validCompanyNumber)
+      cy.get('#company-number-form > button').click()
+      cy.get('h1').should('contain', 'Your payment service provider (PSP) - Stripe')
+    })
+
+    it('should have Company registration number task hyperlink removed when complete and status updated to "COMPLETE " ', () => {
+      setupYourPspStubs({
+        companyNumber: true
+      })
+
+      cy.visit(`/account/${gatewayAccountExternalId}/your-psp/${credentialExternalId}`)
+      cy.get('strong[id="task-Company-number-status"]').should('contain', 'complete')
```

Comment:
Looks like this test should be checking that the hyperlink is removed.

I also noticed in the [story review for the VAT number page](https://payments-platform.atlassian.net/browse/PP-10352) that this assertion is missing there.

Reply:
Yes good spot spot!